semantic_router/indices/local_index.py

```diff
@@ -0,0 +1,37 @@
+import numpy as np
+from typing import List, Any, Tuple, Optional
+from semantic_router.linear import similarity_matrix, top_scores
+from semantic_router.indices.base import BaseIndex
+
+
+class LocalIndex(BaseIndex):
+    index: Optional[np.ndarray] = None
+
+    class Config:  # Stop pydantic from complaining about  Optional[np.ndarray] type hints.
+        arbitrary_types_allowed = True
+
+    def add(self, embeds: List[Any]):
+        embeds = np.array(embeds)  # type: ignore
+        if self.index is None:
+            self.index = embeds  # type: ignore
+        else:
+            self.index = np.concatenate([self.index, embeds])
+
+    def remove(self, indices_to_remove: List[int]):
+        """
+        Remove all items of a specific category from the index.
+        """
+        if self.index is not None:
+            self.index = np.delete(self.index, indices_to_remove, axis=0)
+
+    def is_index_populated(self):
```

Comment:
rather than `is_index_populated` I think we should have something like `.describe_index()` and return the total number of vectors, and dimensionality of the index

Reply:
You've done this for me in `pinecone.py` and `local.py`, I've also made sure it's present in `base.py`.